velox/type/tests/FilterTest.cpp

```diff
@@ -898,3 +898,55 @@ TEST(FilterTest, mergeWithBigintMultiRange) {
     }
   }
 }
+
+TEST(FilterTest, mergeMultiRange) {
+  std::vector<std::unique_ptr<Filter>> filters;
+  addUntypedFilters(filters);
+
+  filters.push_back(
+      orFilter(lessThanOrEqualFloat(-1.2), betweenFloat(1.2, 3.4)));
+  filters.push_back(
+      orFilter(lessThanOrEqualFloat(-1.2), betweenFloat(1.2, 3.4), true));
+
+  filters.push_back(orFilter(lessThanFloat(1.2), greaterThanFloat(3.4)));
+  filters.push_back(orFilter(lessThanFloat(1.2), greaterThanFloat(3.4), true));
+
+  filters.push_back(
+      orFilter(lessThanOrEqualFloat(1.2), greaterThanOrEqualFloat(3.4)));
+  filters.push_back(
+      orFilter(lessThanOrEqualFloat(1.2), greaterThanOrEqualFloat(3.4), true));
+
+  for (const auto& left : filters) {
+    for (const auto& right : filters) {
+      testMergeWithFloat(left.get(), right.get());
+    }
+  }
+}
+
+TEST(FilterTest, cloneTest) {
```

Comment:
Could you explain why all of them are expected to return true for nullallowed?

Reply:
naming: cloneTest -> clone

Reply:
Those filters are constructed by setting `nullAllowed` as `false`. Then I explicitly call `clone(true)` to reset the `nullAllowed` to `true`, and then expect testIsNull returns true.